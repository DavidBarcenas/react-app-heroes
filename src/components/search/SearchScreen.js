import React from 'react'
import { useForm } from '../../hooks/useForm'
import { heroes } from '../../data/heroes'
import { HeroCard } from '../heroes/HeroCard'

export const SearchScreen = () => {

  const {values, handleInputChange} = useForm({
    search: ''
  })

  const heroesFiltered = heroes;

  const handleSearch = (e) => {
    e.preventDefault();
    console.log(values)
  }

  return (
    <div>
      <h1>Search</h1>

      <form onSubmit={handleSearch}>
        <input type="text" placeholder="Find your hero" name="search" onChange={handleInputChange} value={values.search} autoComplete="off" />
        <button type="submit">Search</button>
      </form>

      <div>
        <h4>Results</h4>
        {
          heroesFiltered.map(hero => (
            <HeroCard key={hero.id} hero={hero} />
          ))
        }
      </div>
    </div>
  )
}
