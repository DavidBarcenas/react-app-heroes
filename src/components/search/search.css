.title__section {
  color: #fff;
  font-weight: bold;
  font-size: 25px;
  letter-spacing: 10px;
  text-transform: uppercase;
  display: block;
  text-align: center;
  opacity: 0.65;
}

.search__form {
  display: flex;
  align-items: center;
  width: 40%;
  margin: auto;
}

.search__form input {
  color: #fff;
}

.search__wrapper--card {
  text-align: center;
  margin-top: 2em;
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  justify-content: center;
}

.search__results {
  width: 23%;
  margin: .5em;
}