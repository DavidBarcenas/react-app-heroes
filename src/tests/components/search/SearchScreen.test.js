import '@testing-library/jest-dom';
import React from 'react'
import { mount } from 'enzyme';
import { MemoryRouter, Route } from 'react-router-dom';
import { SearchScreen } from '../../../components/search/SearchScreen';

describe('<SearchScreen /> testing', () => {
  
  test('should render correcly with default values', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['/search']}>
        <Route path='/search' component={SearchScreen}/>
      </MemoryRouter>
    )

    expect(wrapper).toMatchSnapshot()
  }) 
})
