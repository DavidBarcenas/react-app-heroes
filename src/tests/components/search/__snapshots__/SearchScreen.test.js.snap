// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchScreen /> testing should render correcly with default values 1`] = `
<div>
  <span
    className="title__section"
  >
    Search
  </span>
  <form
    className="search__form"
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      name="search"
      onChange={[Function]}
      placeholder="Find your hero"
      type="text"
      value=""
    />
    <button
      className="btn-floating btn-large waves-effect waves-light red"
      type="submit"
    >
      <i
        className="material-icons"
      >
        search
      </i>
    </button>
  </form>
  <div
    className="search__wrapper--card"
  />
</div>
`;
